func_darwin_arm64.go

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2025 The Ebitengine Authors
+
+//go:build darwin && arm64
+
+package purego
+
+import "reflect"
+
+// shouldBundleStackArgs determines if we need to start C-style packing for
+// Darwin ARM64 stack arguments. This happens when registers are exhausted.
+func shouldBundleStackArgs(v reflect.Value, numInts, numFloats int) bool {
```

Comment:
What about moving `shouldBundleStackArgs` and `structFitsInRegisters` to `func.go` and remove `func_darwin_arm64.go` and `func_notdarwin.go`? Adding a check like 

```go
func shouldBundleStackArgs(v reflect.Value, numInts, numFloats int) bool {
    if runtime.GOOS != "darwin" || runtime.GOARCH != "arm64" {
        panic("purego: shouldBundleStackArgs must be called on darwin/arm64")
    }
    // ...
}
```

inside the functions would be nice.

Reply:
could do struct_nondarwin.go, struct_darwin_arm64.go, struct_darwin_amd64.go if we don't want to propagate func_* files.